unic/ucd/normal/src/decomposition_type.rs

```diff
@@ -12,228 +12,180 @@
+        Canonical {
+            abbr => Can,
+            long => Canonical,
+            human => "Canonical",
+        }
 
-/// Represents the Unicode character
-/// [*Decomposition_Type*](http://www.unicode.org/reports/tr44/#Decomposition_Type) property.
-///
-/// * <http://www.unicode.org/reports/tr44/#Character_Decomposition_Mappings>
-#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
-#[allow(missing_docs)]
-pub enum DecompositionType {
-    Canonical,
-    Compat,
-    Circle,
-    Final,
-    Font,
-    Fraction,
-    Initial,
-    Isolated,
-    Medial,
-    Narrow,
-    Nobreak,
-    None,
-    Small,
-    Square,
-    Sub,
-    Super,
-    Vertical,
-    Wide,
-}
-
+        /// Otherwise unspecified compatibility character
+        Compat {
+            abbr => Com,
+            long => Compat,
+            human => "Unspecified",
+        }
 
-impl CharProperty for DecompositionType {
-    fn prop_abbr_name() -> &'static str {
-        "dt"
-    }
+        /// Encircled form
+        Circle {
+            abbr => Enc,
+            long => Circle,
+            human => "Encircled",
+        }
 
-    fn prop_long_name() -> &'static str {
-        "Decomposition_Type"
-    }
+        /// Final presentation form (Arabic)
+        Final {
+            abbr => Fin,
+            long => Final,
+            human => "Arabic Final",
+        }
 
-    fn prop_human_name() -> &'static str {
-        "Decomposition Type"
-    }
-}
+        /// Font variant (for example, a blackletter form)
+        Font {
+            abbr => Font,
+            long => Font,
+            human => "Font Variant",
+        }
 
+        /// Vulgar fraction form
+        Fraction {
+            abbr => Fra,
+            long => Fraction,
+            human => "Vulgar Fraction",
+        }
 
-impl PartialCharProperty for DecompositionType {
-    fn of(ch: char) -> Option<Self> {
-        Self::of(ch)
-    }
-}
+        /// Initial presentation form (Arabic)
+        Initial {
+            abbr => Init,
+            long => Initial,
+            human => "Arabic Initial",
+        }
 
+        /// Isolated presentation form (Arabic)
+        Isolated {
+            abbr => Iso,
+            long => Isolated,
+            human => "Arabic Isolated",
+        }
 
-impl EnumeratedCharProperty for DecompositionType {
-    fn all_values() -> &'static [Self] {
-        Self::all_values()
-    }
+        /// Medial presentation form (Arabic)
+        Medial {
+            abbr => Med,
+            long => Medial,
+            human => "Arabic Medial",
+        }
 
-    fn abbr_name(&self) -> &'static str {
-        self.abbr_name()
-    }
+        /// Narrow (or hankaku) compatibility character
+        Narrow {
+            abbr => Nar,
+            long => Narrow,
+            human => "Narrow",
+        }
 
-    fn long_name(&self) -> &'static str {
-        self.long_name()
-    }
+        /// No-break version of a space or hyphen
+        NoBreak {
```

Comment:
I'm pretty sure the UCD will never change the representation in a file to support old parsers for as long as possible. By the spec, comparison of property keys and values should be case insensitive and ignore `_` and ` `. (So "`NOBREAK`", "`NO_BREAK`", "`n  o _    _ bre  ____AK`" are all equivalent values.)

In #58 we discussed a similar case with naming `ccc=204` and the other classes with a description with no name. There we reached a rough consensus at

> @cad97
> The guideline I used here was basically  
> "The Unicode document provides a description for the (unnamed) value, which follows a pattern of descriptions for other (named) values, from which the description => name transformation can be trivially derived"
>
> @behnam 
> Sound reasonable, specially since it's not the [name in the UCD name alias module], but a local naming based on the [UCD name].

I think we can expand that to slightly modifying slightly-out-of-line UCD naming by a similar rule:

> [PROPOSED]
> Naming of enum variants (for `EnumeratedCharProperty`s) and variant-like constants (for other properties) follows Rust naming conventions for enum variants (namely, `CamelCase`). Namely, Unicode "Long Aliases" of the format `Long_Name` (sourced from PropertyValueAliases) will have their underscores removed to create a `CamelCase` identifier. If other appearances of the data in normative sources use a form not listed in PropertyValueAliases.txt, the "word breaking" (in whatever form: `snake_case`, `lowerCamel`, `CamelCase`, `SCREAMING_CASE` `Sentence structure`) of that format will be considered; and if it is deemed to have more natural breaks, those will be used instead.